tensorflow/python/ops/distributions/multinomial.py

```diff
@@ -260,7 +261,7 @@ def _log_prob(self, counts):
 
   def _log_unnormalized_prob(self, counts):
     counts = self._maybe_assert_valid_sample(counts)
-    return math_ops.reduce_sum(counts * math_ops.log(self.probs), -1)
+    return -nn_ops.softmax_cross_entropy_with_logits(labels=counts, logits=self.logits)
```

Comment:
there is a known bug wherein softmax_cross_entropy_with_logits does not backprop into its labels argument; so you cannot backprop into counts using this function.